addons/binding/org.openhab.binding.chamberlainmyq/src/main/java/org/openhab/binding/chamberlainmyq/handler/ChamberlainMyQGatewayHandler.java

```diff
@@ -0,0 +1,472 @@
+    }
+
+    /**
+     * Retrieves MyQ device data from myq website, throws if connection
+     * fails or user login fails
+     *
+     */
+    public MyqJson getMyqData() throws InvalidLoginException, IOException {
+        logger.debug("Retrieving door data");
+        String tempToken = getSecurityToken();
+        String url = String.format("%s/api/v4/UserDeviceDetails/Get?appId=%s&SecurityToken=%s", WEBSITE, enc(APP_ID),
+                enc(tempToken));
+
+        JsonObject data = request("GET", url, null, null, true, false, enc(tempToken));
+        return gson.fromJson(data, MyqJson.class);
+    }
+
+    /**
+     * Send Command to open/close garage door opener with MyQ API Returns false
+     * if return code from API is not correct or connection fails
+     *
+     * @param deviceID MyQ deviceID of Garage Door Opener.
+     * @param name     Attribute Name "desireddoorstate" or "desiredlightstate"
+     * @param state    Desired state to put the door in, 1 = open, 0 = closed
+     *                     Desired state to put the lamp in, 1 = on, 0 = off
+     */
+    public void executeMyQCommand(String deviceID, String name, int state, boolean rapidPoll)
+            throws InvalidLoginException, IOException {
+        String tempToken = getSecurityToken();
+        String message = String.format(
+                "{\"ApplicationId\":\"%s\"," + "\"SecurityToken\":\"%s\"," + "\"MyQDeviceId\":\"%s\","
+                        + "\"AttributeName\":\"%s\"," + "\"AttributeValue\":\"%d\"}",
+                APP_ID, enc(tempToken), deviceID, name, state);
+        String url = String.format("%s/api/v4/DeviceAttribute/PutDeviceAttribute?appId=%s&SecurityToken=%s", WEBSITE,
+                enc(APP_ID), enc(tempToken));
+
+        request("PUT", url, message, "application/json", true, false, enc(tempToken));
+
+        if (rapidPoll) {
+            beginRapidPoll();
+        } else {
+            doFuturePoll(quickPoll * 1000);
+        }
+
+    }
+
+    public interface RequestCallback {
+        public void parseRequestResult(MyqJson resultJson);
+
+        public void onError(String error);
+    }
+
+    protected class Request implements Runnable {
+        private RequestCallback callback;
+
+        public Request(RequestCallback callback) {
+            this.callback = callback;
+        }
+
+        @Override
+        public void run() {
+            try {
+                MyqJson resultJson = getMyqData();
+                callback.parseRequestResult(resultJson);
+            } catch (Exception e) {
+                logger.warn("An exception occurred while executing a request to the MyQ Gateway: '{}'", e.getMessage());
+            }
+        }
+    }
+
+    public void sendRequestToServer(RequestCallback callback) throws IOException {
+        Request request = new Request(callback);
+        request.run();
+    }
+
+    private void startAutomaticRefresh() {
+        mainPollRefreshJob = scheduler.scheduleAtFixedRate(() -> {
```

Comment:
Please use [scheduleWithFixedDelay](http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleWithFixedDelay(java.lang.Runnable,%20long,%20long,%20java.util.concurrent.TimeUnit)) which is preferred over scheduleAtFixedRate, see [Coding Guidelines D.3](http://docs.openhab.org/developers/development/guidelines.html#d-runtime-behavior)